docs/express.asciidoc

```diff
@@ -0,0 +1,250 @@
+[[express]]
+== Getting started with Express
+
+Getting Elastic APM set up for your Express app is easy,
+and there are various ways you can tweak it to fit to your needs.
+
+Follow the guide below and refer to the <<api,API documentation>> for all the advanced stuff.
+
+[[express-installation]]
+=== Installation
+
+Add the `elastic-apm` module as a dependency to your application:
+
+[source,bash]
+----
+npm install elastic-apm --save
+----
+
+[[express-initialization]]
+=== Initialization
+
+It's important that the agent is started before you require *any* other modules in your Node.js application - i.e. before `express`, `http`, etc.
+
+This means that you should probably require and start it in your applications main file (usually `index.js`, `server.js` or `app.js`).
+
+Here's a simple Express example with the Elastic APM agent installed:
+
+[source,js]
+----
+// Add this to the VERY top of the first file loaded in your app
+var apm = require('elastic-apm').start({
+  // Set required app name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
+  appName: '',
+
+  // Use if APM Server requires a token
+  secretToken: '',
+
+  // Set custom APM Server URL (default: http://localhost:8200)
+  serverUrl: '',
+})
+
+var app = require('express')()
+
+app.get('/', function (req, res) {
+  res.send('Hello World!')
+})
+
+// any errors caught by Express can be logged by the agent as well
+app.use(apm.middleware.express())
+
+app.listen(3000)
+----
+
+The agent will now monitor the performance of your Express application and record any uncaught exceptions.
+
+[[express-advanced-configuration]]
+==== Advanced configuration
```

Comment:
You should consider including shared files for content that is duplicated in multiple places. Let me know if you need help figuring out how to do this.

Reply:
Some of these shared sections are only 99% identical. The only difference is that they use the name of the framework - so one says "Koa" and another says "Express" - but besides that they are the same. I was thinking if it was possible to use something like `{framework}` to generalise it?

Reply:
Yes, that'll work. Put each chunk of shared content in a separate file and use the `{framework}` attribute to set the framework name. Use includes to pull in the shared content. Then set the value of `framework` in the index file. For example:

```
:framework: Express
include::./express.asciidoc[]

:framework: Koa
include::./koa.asciidoc[] 
```

For larger sections (like code examples), you can can use conditional coding (let me know if you need to do that).

Reply:
Awesome, thank you 😄 I need to clean up some things in the Express part to make it identical (requires code changes as well), so I'll wait until that's done to do this.